xarray/tests/test_datatree.py

```diff
@@ -666,20 +690,229 @@ def test_repr(self):
             """
             <xarray.DataTree 'b'>
             Group: /b
-            │   Dimensions:  (y: 1)
-            │   Dimensions without coordinates: y
+            │   Dimensions:  (x: 2, y: 1)
+            │   Coordinates:
+            │     * x        (x) float64 16B 2.0 3.0
             │   Data variables:
             │       f        (y) float64 8B 3.0
             ├── Group: /b/c
             └── Group: /b/d
-                    Dimensions:  ()
-                    Data variables:
+                    Dimensions:  (x: 2, y: 1)
+                    Coordinates:
+                      * x        (x) float64 16B 2.0 3.0
+                    Dimensions without coordinates: y                    Data variables:
                         g        float64 8B 4.0
             """
         ).strip()
         assert result == expected
 
 
+class TestInheritance:
+    @pytest.mark.xfail(reason="I don't understand the expected behaviour here")
+    def test_inherited_dims(self):
+        dt = DataTree.from_dict(
+            {
+                "/": xr.Dataset(data_vars={"d": (("x",), [1, 2])}),
+                "/b": xr.Dataset(data_vars={"e": (("y",), [3])}),
+                "/c": xr.Dataset(data_vars={"f": (("y",), [3, 4, 5])}),
+            }
+        )
+        assert dt.sizes == {"x": 2}
+        # nodes should include inherited dimensions
+        assert dt["b"].sizes == {"x": 2, "y": 1}
+        assert dt["c"].sizes == {"x": 2, "y": 3}
```

Comment:
@shoyer why would I expect these dimensions to be present if the parent node only has them on data variables, not coordinates?